external/kafka/src/main/scala/org/apache/spark/streaming/kafka/DirectKafkaInputDStream.scala

```diff
@@ -158,4 +166,37 @@ class DirectKafkaInputDStream[
     }
   }
 
+  private[streaming]
+  class DirectKafkaStreamingListener extends StreamingListener {
+
+    val offsetCommitEnabled = context.conf.getBoolean("spark.streaming.kafka.offsetCommit", false)
+
+    val zkClientOpt = if (offsetCommitEnabled) {
+      kafkaParams.get("zookeeper.connect").map { zkConnect =>
```

Comment:
Doing things this way now means that zookeeper connect is required in the kafka params, whereas it was not previously.

Why not use the existing setConsumerOffset methods in KafkaCluster?  That also means that this could more easily be updated to work with Kafka-based consumer offset storage post-0.8.2